mlir/lib/Dialect/GPU/Transforms/ModuleToBinary.cpp

```diff
@@ -48,9 +48,7 @@ void GpuModuleToBinaryPass::getDependentDialects(
   // Register all GPU related translations.
   registry.insert<gpu::GPUDialect>();
   registry.insert<LLVM::LLVMDialect>();
-#if LLVM_HAS_NVPTX_TARGET
   registry.insert<NVVM::NVVMDialect>();
-#endif
 #if MLIR_ENABLE_ROCM_CONVERSIONS
   registry.insert<ROCDL::ROCDLDialect>();
 #endif
```

Comment:
Do we need this registration calls? As far as I can tell, no, all these dialects should already be in the context by the time this pass gets invoked.

Reply:
The LLVM dialect may not be?

Reply:
Actually it's unlikely, I'll remove them all and wait for a test failure!

Reply:
The only op generated by the pass is a `gpu.binary`, and `gpu` should be in the context because the pass only performs serialization if there's a `gpu.module` in the input.

All other dialects, `llvm`, `nvvm`, `rocdl`, `spirv`, have to be in the input or loaded by a `register*Translation*` call. And all the `register*Translation*` have to be called before invoking the pass.

Reply:
Quick question, let's suppose there are no GPU modules or ops, does `dyn_cast<gpu::GPUModuleOp>` requires the dialect to be in the context?
If it does, then we do need `registry.insert<gpu::GPUDialect>` and that's the only one we might need.

Reply:
> Quick question, let's suppose there are no GPU modules or gpu ops, does dyn_cast<gpu::GPUModuleOp> requires the dialect to be in the context?

No it does not.